### PR TITLE
Document Consul ECS Resource Usage

### PR DESCRIPTION
**Pull request by erichaberkorn**

### CODE REVIEW

File: website/content/docs/ecs/resource-usage.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+layout: docs
+page_title: Resource Usage - AWS ECS
+description: >-
+  Resource usage of the Consul Service Mesh on AWS ECS (Elastic Container Service).
+---
+
+# Resource Usage
+
+We ran performance tests on Consul on ECS to determine the resource usage of
```
Comment (pglass) [inline website/content/docs/ecs/resource-usage.mdx:10]:
```suggestion
We ran performance tests of Consul on ECS to determine the resource usage of
```

File: website/content/docs/ecs/resource-usage.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+the sidecar containers Consul on ECS injects along with the ACL controller.
+[The architecture page](/docs/ecs/architecture) describes each Consul on ECS
+component in depth.
+
+We ran performance tests by deploying clusters of varying sizes and measuring
+resource usage during conditions that stress Consul on ECS components. After
+each iteration of performance testing, we recorded resource usages of each
+component to determine worst-case scenario resource usage in a production
+environment. During performance testing, we used ECS' minimum allowable CPU and memory
+resources, 256 CPU shares and 512 MB of memory, to demonstrate that
```
Comment (pglass) [inline website/content/docs/ecs/resource-usage.mdx:20]:
```suggestion
environment. During performance testing, we used Fargate's minimum allowed CPU and memory
settings on ECS, 256 CPU shares and 512 MB of memory, to demonstrate that
```

nitpicking: the EC2 launch type doesn't restrict you to the Fargate cpu/memory configurations

File: website/content/docs/ecs/resource-usage.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+tasks. Here is the maximum resource usage we observed for each container:
+
+| Container      | CPU | Memory |
+| -------------- | --- | ------ |
+| ACL Controller | 5%  | 43 MB  |
+| Health Sync    | 6%  | 35 MB  |
+| Consul Client  | 10% | 87 MB  |
+| Envoy          | 18% | 55 MB  |
+
+As shown above, Consul on ECS’ containers use well under the minimum CPU and
```
Comment (pglass) [inline website/content/docs/ecs/resource-usage.mdx:31]:
```suggestion
The containers added by Consul on ECS use well under the minimum CPU and
```

File: website/content/docs/ecs/resource-usage.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+component in depth.
+
+We ran performance tests by deploying clusters of varying sizes and measuring
+resource usage during conditions that stress Consul on ECS components. After
+each iteration of performance testing, we recorded resource usages of each
+component to determine worst-case scenario resource usage in a production
+environment. During performance testing, we used ECS' minimum allowable CPU and memory
+resources, 256 CPU shares and 512 MB of memory, to demonstrate that
+Consul on ECS along with application containers can run on the smallest ECS
+tasks. Here is the maximum resource usage we observed for each container:
```
Comment (pglass) [inline website/content/docs/ecs/resource-usage.mdx:22]:
```suggestion
tasks. The following table shows the maximum resource usage we observed for each container:
```

One of my doc learnings from Adam is to make the document read sort of linearly downward. So more explaining what follows, and less "as shown above" where the reader has to jump back up.

File: website/content/docs/ecs/resource-usage.mdx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+| Container      | CPU | Memory |
+| -------------- | --- | ------ |
+| ACL Controller | 5%  | 43 MB  |
+| Health Sync    | 6%  | 35 MB  |
+| Consul Client  | 10% | 87 MB  |
+| Envoy          | 18% | 55 MB  |
+
+As shown above, Consul on ECS’ containers use well under the minimum CPU and
+memory limits for an ECS Task. `memory` and `cpu` can be passed into Mesh Init
+if additional resources are necessary.
```
Comment (pglass) [inline website/content/docs/ecs/resource-usage.mdx:33]:
```suggestion
memory limits for an ECS Task. Use the `memory` and `cpu` settings for the task definition
if additional resources are necessary for your application task.
```

Clarifying slightly that they should only need additional cpu/memory for the Consul ECS containers based on their application's needs.

File: website/data/docs-nav-data.json [hunk 6]
```diff
@@ -615,6 +615,10 @@
         "title": "Requirements",
         "path": "ecs/requirements"
       },
+      {
+        "title": "Resource Usage",
+        "path": "ecs/resource-usage"
```
Comment (pglass) [inline website/data/docs-nav-data.json:620]: What do you think about instead adding this as a "Task Resource Usage" section in the Requirements page?

I'm good with it either way.

**Review by pglass — APPROVED** [review]
Looks good! Some minor wording suggestions (feel free to use your own alternatives)

File: website/content/docs/ecs/resource-usage.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  Resource usage of the Consul Service Mesh on AWS ECS (Elastic Container Service).
+---
+
+# Resource Usage
+
+We ran performance tests on Consul on ECS to determine the resource usage of
+the sidecar containers Consul on ECS injects along with the ACL controller.
+[The architecture page](/docs/ecs/architecture) describes each Consul on ECS
+component in depth.
+
```
Comment (trujillo-adam) [inline website/content/docs/ecs/resource-usage.mdx:14]: Would the paragraph starting on line15 be easier to consume in list format, i.e.:

We used the following procedure to measure resource usage:
* Executed performance tests while deploying clusers of various sizes. We ensured that deployment conditions stressed Consul on ESC components.
* After each performance test session, we recorded resource usage for each component to determine worst-case scenario resource usage in a production environment. 
* We used Fargate's minimum allowed CPU (256 shares) and memory settings (512 MB) on ECS during performance testing to demonstrate that Consul on ECS along with application containers can run on the smallest ECS tasks.

File: website/content/docs/ecs/resource-usage.mdx (hunk 3 above)
Comment (trujillo-adam) [reply website/content/docs/ecs/resource-usage.mdx:31]:
```suggestion
The containers added by Consul on ECS consume resources well below the minimum CPU and
```
I'm not sure if that changes the meaning - in any case, "below" is probably the correct diction.

**Review by trujillo-adam — COMMENTED** [review]
Left a few questions/suggestions

**Review by pglass — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/593440.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a16f665f0f70d7a59d4bf212d1102a4941b1fcb8 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a16f665f0f70d7a59d4bf212d1102a4941b1fcb8 onto `release/1.11.x` succeeded!

**Pull request merged**